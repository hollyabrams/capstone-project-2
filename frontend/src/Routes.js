import { Route, Switch } from 'react-router-dom';
import Homepage from './Homepage';
import ProductList from './components/ProductList';
import LoginForm from './forms/LoginForm';
import SignupForm from './forms/SignupForm';
import ProfileForm from './forms/ProfileForm';
import ProductPage from './ProductPage';
import CartPage from './CartPage';
import SuccessPage from './Success';

const Routes = ({ login, signup }) => {
  return (
    <>
      <Switch>
        <Route exact path="/">
          <Homepage />
        </Route>
        <Route exact path="/products">
          <ProductList />
        </Route>
        <Route exact path="/products/:productId">
          <ProductPage />
        </Route>
        <Route exact path="/login">
          <LoginForm login={login} exact path="/login" />
        </Route>
        <Route exact path="/signup">
          <SignupForm signup={signup}/>
        </Route> 
        <Route exact path="/profile">
          <ProfileForm />
        </Route>
        <Route exact path="/cart">
          <CartPage />
        </Route>
        <Route exact path="/success">
          <SuccessPage />
        </Route>
      </Switch>
    </>
  );
};

export default Routes;
