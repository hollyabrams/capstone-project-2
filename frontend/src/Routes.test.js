import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Routes from './Routes';


jest.mock('./Homepage', () => () => <div>Homepage</div>);
jest.mock('./components/ProductList', () => () => <div>ProductList</div>);
jest.mock('./forms/LoginForm', () => () => <div>LoginForm</div>);
jest.mock('./forms/SignupForm', () => () => <div>SignupForm</div>);
jest.mock('./forms/ProfileForm', () => () => <div>ProfileForm</div>);
jest.mock('./ProductPage', () => () => <div>ProductPage</div>);
jest.mock('./CartPage', () => () => <div>CartPage</div>);
jest.mock('./Success', () => () => <div>SuccessPage</div>);

describe('Routes component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <Routes />
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/']}>
        <Routes />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
