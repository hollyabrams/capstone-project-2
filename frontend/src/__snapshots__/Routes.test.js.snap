// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Homepage
  </div>
</DocumentFragment>
`;
